# HG changeset patch
# User Rik <rik@octave.org>
# Date 1546494296 28800
#      Wed Jan 02 21:44:56 2019 -0800
# Branch stable
# Node ID 9ec0a1e5d4d3c9edecb2691a7945175ca56f6529
# Parent  3774259dd1146e14686e1a579d24e14b66cd41a9
__ode15__.cc: Fix static analyzer detected issues (bug #55347).

* __ode15__.cc (F__ode15__): Declare and assign *ida_fcn in one statement.
Use comparison to UNSORTED rather than "! issorted ()" since return value is
an enum, not bool.

diff --git a/libinterp/dldfcn/__ode15__.cc b/libinterp/dldfcn/__ode15__.cc
--- a/libinterp/dldfcn/__ode15__.cc
+++ b/libinterp/dldfcn/__ode15__.cc
@@ -1127,36 +1127,34 @@ Undocumented internal function.
 
   // Check number of parameters
   int nargin = args.length ();
 
   if (nargin != 5)
     print_usage ();
 
   // Check odefun
-  octave_function *ida_fcn = nullptr;
-
   octave_value f_arg = args(0);
 
   if (! f_arg.is_function_handle ())
     error ("__ode15__: odefun must be a function handle");
 
-  ida_fcn = f_arg.function_value ();
+  octave_function *ida_fcn = f_arg.function_value ();
 
   // Check input tspan
   ColumnVector tspan
     = args(1).xvector_value ("__ode15__: TRANGE must be a vector of numbers");
 
   int numt = tspan.numel ();
 
   realtype t0 = tspan (0);
 
   if (numt < 2)
     error ("__ode15__: TRANGE must contain at least 2 elements");
-  else if (! tspan.issorted () || tspan(0) == tspan(numt - 1))
+  else if (tspan.issorted () == UNSORTED || tspan(0) == tspan(numt - 1))
     error ("__ode15__: TRANGE must be strictly monotonic");
 
   // input y0 and yp0
   ColumnVector y0
     = args(2).xvector_value ("__ode15__: initial state y0 must be a vector");
 
   ColumnVector yp0
     = args(3).xvector_value ("__ode15__: initial state yp0 must be a vector");
